torch/_inductor/fx_passes/kernel_optimization.py

```diff
@@ -0,0 +1,74 @@
+# mypy: allow-untyped-defs
+import logging
+from typing import Optional
+
+import torch
+
+from torch._dynamo.utils import counters, is_node_meta_valid
+
+from .. import config
+
+from ..kernel.quack.quack_rmsnorm import rmsnorm as quack_rmsnorm
+from ..pattern_matcher import (
+    CallFunctionVarArgs,
+    Match,
+    MULTIPLE,
+    register_graph_pattern,
+)
+from .split_cat import construct_pattern_matcher_pass
+
+log = logging.getLogger(__name__)
+
+
+def check_device(inputs: list[torch.Tensor], device="cuda") -> bool:
+
+    return all(input.device.type == device for input in inputs)
+
+
+def should_replace_norm(inputs: list[Optional[torch.fx.Node]]) -> bool:
+    inputs = [input for input in inputs if input is not None]
+
+    if not all(is_node_meta_valid(input) for input in inputs):
+        return False
+
+    return check_device([input.meta["example_value"] for input in inputs])
+
+
+def print_norm_pattern(match: Match, inputs: list[Optional[torch.fx.Node]]):
+    node = match.nodes[-1]
+    log.debug(
```

Comment:
This is still eagerly does the string join, either wrap it in a LazyStr or put it in a proper debugging level if